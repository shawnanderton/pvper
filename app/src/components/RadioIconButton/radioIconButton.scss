$selected-color: #e02929;

.radio-icon-button {
  padding: 0;
  margin: 0;
  border-radius: 3px;
  border: none;
  background-color: inherit;
}

.selected {
  border: solid 1px $selected-color;
}
