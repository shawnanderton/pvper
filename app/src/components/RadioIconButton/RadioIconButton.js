import React, { useState } from 'react';

import Icon from '../Icon';

import './radioIconButton.scss';

function RadioIconButton({
  id,
  selected = false,
  icon = { name: '', size: 24 },
  onChange,
}) {
  const handleClick = () => {
      console.log(selected);
    onChange({ key: id, selected: !selected });
  };
  return (
    <button
      onClick={handleClick}
      className={`radio-icon-button ${selected ? 'selected' : ''}`}
    >
      <Icon name={icon.name} size={icon.size} />
    </button>
  );
}

export default RadioIconButton;
