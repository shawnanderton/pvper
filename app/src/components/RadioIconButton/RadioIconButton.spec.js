import React from 'react'
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import RadioIconButton from './RadioIconButton';

describe('RadioIconButton', () => {

     it('should render correctly', () => {
    const tree = renderer
      .create(
        <RadioIconButton  />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have Icon set', () => {
      const wapper =   shallow(<RadioIconButton />);
      const icon =  wapper.find('Icon');

        expect(icon.length).toBe(1);     
  });
   it('should have selected class style if selected is true', () => {
       const icon ={name:"horde", size:24};
      const wapper =   shallow(<RadioIconButton selected={true} icon={icon} />);
      const buttonSelected =  wapper.find('button.selected');

        expect(buttonSelected.length).toBe(1);     
  });

   it('should not have selected class style if selected is false', () => {
        const icon ={name:"horde", size:24};
      const wapper =   shallow(<RadioIconButton selected={false}  icon={icon} />);
      const buttonSelected =  wapper.find('button.selected');

        expect(buttonSelected.length).toBe(0);     
  });
});