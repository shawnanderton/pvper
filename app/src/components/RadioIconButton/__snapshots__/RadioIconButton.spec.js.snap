// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioIconButton should render correctly 1`] = `
<button
  className="radio-icon-button "
  onClick={[Function]}
>
  <button
    aria-describedby={null}
    className="MuiButtonBase-root"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    title=""
    type="button"
  >
    <img
      alt=""
      style={
        Object {
          "height": 24,
          "width": 24,
        }
      }
    />
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</button>
`;
